config/samples/all-in-one/kustomization.yaml

```diff
@@ -1,5 +1,18 @@
 ## Append samples you want in your CSV to this file as resources ##
+generatorOptions:
+ disableNameSuffixHash: true
+
+secretGenerator:
+- name: kbs-auth-public-key
+  files:
+  - kbs.pem
+#- name: kbsres1
+#  literals:
+#  - key1=res1val1
+#  - key2=res1val2
+
```

Comment:
Can you try adding a patch.yaml like
```
apiVersion: confidentialcontainers.org/v1alpha1
kind: KbsConfig
metadata:
  name: kbsconfig-sample
spec:
  kbsSecretResources:
    - "kbsres1"
```

and include it in kustomization.yaml like
```
[snip]

secretGenerator:
- name: kbs-auth-public-key
  files:
  - kbs.pem
#- name: kbsres1
#  literals:
#  - key1=res1val1
#  - key2=res1val2

#patches:
#- patch.yaml

resources:
- kbsconfig_sample.yaml
[snip]
```
So when you uncomment the secret and the patch, and apply it the kbsconfig should have the kbsSecretResources.
We can then use this model for the optional elements like rvps values as well.

Reply:
I see... then you can mention about having a valid json file in the readme and uncommenting relevant entries in kustomization to use the json file as configmap

```
configmapGenerator:
- name: ..
  namespace: ..
  files:
  #- rvps-ref-values.json
```